### ISSUE TITLE
EPIC v0.5.0 — Parity with whisper.cpp

### ISSUE THREAD

**Issue by noahgift** [paiml/whisper.apr#90]
Epic of the whisper.apr release plan, drafted 2026-09-23 by aprender-cb (the standing whisper.apr owner) from the operator's request, relayed verbatim by the cop (aprender-cf): *"we need to add whisper.apr too and xpile"* / *"ensure all of these have next five epics planned out as they will be continuously worked on"*. It follows apex's pattern (paiml/apex#130–#134): **one milestone per tag, one epic per milestone**, and every row carries a `done_when` that can fail. **Operator review is pending, and nothing is reordered before it.** A 3-lane non-Claude grill of this plan runs before any row moves between releases.

**Facts every epic stands on** (measured 2026-09-23 on `origin/main` @ `d46f381`, unless dated otherwise):
- **Versioning: two release lines share one tag namespace.** (1) The **crate**: crates.io `whisper-apr` **0.3.3** (2026-08-15) has **no git tag and no GitHub release**; the newest crate-line GitHub release is v0.3.2. (2) The **desktop/WASM app**: GitHub release **`v1.0.0` "Whisper.apr v1.0.0 - WASM Speech Recognition"**, published 2026-01-05, with a macOS `.dmg` and a `whisper-tiny-fb.apr` asset (5 downloads each). Its tag sits on a commit that is **not an ancestor of `main`**, which is 500 commits ahead. *Correction: an earlier draft of this epic called `v1.0.0` a "stray tag". It is a published release, and it is left untouched.* This plan's tags continue the **crate** line, **v0.4.0 → v0.8.0**. How the two lines should be named from here (e.g. `app-v*` vs `v*`) is an **operator decision**, recorded in v0.4.0 R1.
- **The CRUX baseline is already RED, and stale.** `docs/specifications/whisper-apr-cpp-parity.md` (dated 2026-02-12) records WER vs whisper.cpp, greedy to greedy: **0% (1.5 s clip), 22% (3 s), 32% (33.6 s)**, against its own target of **≤ 10% (≤ 1% goal)**. It attributes the gap to fp32 SafeTensors vs fp16 ggml weights. Chunking has changed since (#83, #86), so these numbers are **not current**; v0.4.0 R3 re-measures them.
- **The clip set is partly tracked.** In `demos/test-audio/` (31 tracked files): `test-speech-1.5s.wav` (with a `.wav.txt` transcript), `test-speech-3s.wav` (with `.wav.json`), `test-30s.wav`, `silence-5s.wav`, `whisper-5s.wav`, and format and edge variants. **Not tracked:** the spec's 33.6 s `test-speech-full.wav`; `jfk.wav` lives in whisper.cpp's `samples/`. The models (`models/*.apr`, 1.6 GB locally) are not tracked either. *Correction: an earlier draft of this epic said the spec's `test-speech-*` clips were untracked. Two of the three are tracked; a truncated `git ls-files | head` hid them.*
- **The ground-truth tests self-skip.** `tests/ground_truth_tests.rs::load_model_and_audio()` returns `Option`, so the tests pass wherever the model is absent, **CI included**. That is a green that proves nothing.
- The whisper.cpp comparator exists locally (`~/src/whisper.cpp` @ `2551e4ce`, with `whisper-cli` and ggml models), but is **not pinned** anywhere in this repo.

## v0.5.0: "Parity with whisper.cpp"

The spec's own target, made a gate: **WER vs whisper.cpp ≤ 10% on every clip** (greedy to greedy), on the R3 baseline's clip set.

| row | done_when | baseline | depends on |
|---|---|---|---|
| **R1 First-divergence localization** | under **forced alignment** (whisper.cpp's tokens fed to whisper.apr), a per-step table of apr vs cpp top-2 logits on each divergent clip, with the first layer that diverges named (the method of whisper-apr-cpp-parity.md §3.3, rerun) | Feb: "only 2/12 steps diverge independently" (3 s clip) | v0.4.0 R2/R3 |
| **R2 fp16 weight path** | whisper.apr loads f16 weights (`models/*-f16.apr`, or ggml fp16 directly) and computes in the precision the spec names as the root cause; R1's divergences shrink or vanish | fp32 SafeTensors only (spec §3.3) | R1 |
| **R3 WER ≤ 10%** | the CRUX receipt: every clip ≤ 10% WER vs whisper.cpp, on both CPU hosts | 22% / 32% (Feb; re-measured in v0.4.0) | R2 |
| **R4 A second oracle** | an HF transformers Whisper leg in the CRUX harness (the spec cites it as the second reference), so a clip where whisper.cpp itself is wrong is distinguishable | whisper.cpp only | v0.4.0 R2 |

### Release gates
| gate | command | positive control (must go RED) |
|---|---|---|
| **CRUX** | `scripts/crux_wer.sh`: every clip ≤ 10% WER vs whisper.cpp; **no clip worse than v0.4.0's receipt** | planted reference error gives WER > 0; whisper.cpp self-check gives 0%; a fixture with **fp32 weights forced** reproduces the known 22% (proof the gate can see the regression it guards) |
| **Dogfood** | `scripts/release-dogfood.sh 0.5.0` | unpublished version REFUSED |
| **Proving command** | `scripts/crux_wer.sh --clips crux/clips.toml --engines whisper-apr,whisper.cpp,hf --out crux/receipts/0.5.0/` | |
